dep0/lib/01_ast/include/dep0/ast/ast.hpp

```diff
@@ -39,35 +60,63 @@ struct type_t
     struct u16_t { bool operator==(u16_t const&) const { return true; } };
     struct u32_t { bool operator==(u32_t const&) const { return true; } };
     struct u64_t { bool operator==(u64_t const&) const { return true; } };
-    struct name_t
+    struct var_t
     {
-        source_text name;
-        bool operator==(name_t const&) const = default;
+        indexed_var_t name;
+        bool operator<(var_t const& that) const { return name < that.name; }
+        bool operator==(var_t const&) const = default;
+    };
+    struct arr_t
+    {
+        // in lambda-2, an arrow can either introduce new type variables (pi-types) or refer to existing types
+        using arg_kind_t = std::variant<var_t, type_t>;
+        using arg_kinds_iterator = std::vector<arg_kind_t>::iterator;
+        using arg_kinds_const_iterator = std::vector<arg_kind_t>::const_iterator;
+        std::vector<arg_kind_t> arg_kinds;
```

Comment:
Should try with this intead:
```suggestion
        using arg_kind_t = std::variant<var_t, type_t>;
        struct arg_t
        {
            sort_t<P> sort;
            var_t var;
        };
        using arg_iterator = std::vector<arg_t>::iterator;
        using arg_const_iterator = std::vector<arg_t>::const_iterator;
        std::vector<arg_t> args;
```

Reply:
then the commend would be superflous